rustler/src/resource.rs

```diff
@@ -145,6 +145,10 @@ where
         }
     }
 
+    pub fn as_c_arg(&mut self) -> *const c_void {
```

Comment:
Does this really need to be `pub` or is this a debugging residue?

Reply:
We need this public if you want to allow someone to call the wrapper functions directly. I would appreciate that, because we are calling this hundreds of times, sometimes millions of times in a row. And small changes are showing up, especially if we make ResourceArc.make_resource_binary return a Binary like you suggested below. :)

Reply:
At the very least it has to be `unsafe`.

Reply:
Actually, strike that, I don't think we do that on other functions either and you have to use `unsafe` functions to actually do something with the pointer.